src/lights.c

```diff
@@ -430,32 +442,35 @@ s32 get_light_count(void) {
     return gNumActiveLights;
 }
 
-void func_80032C7C(Object *object) {
+/**
+ * Updates the shading on an object based on nearby high-intensity light sources.
+ */
+void update_object_shading(Object *object) {
     ObjectLight *light;
     s16 primIndex, secIndex;
     f32 intensity;
     s16 objX;
     s16 objY;
     s16 objZ;
-    u8 sp64;
+    u8 objTypeMask;
     s32 i;
 
-    if (object->header->unk3D == 0) {
+    if (object->header->shadingIntensity == 0) {
         switch (object->header->modelType) {
             case OBJECT_MODEL_TYPE_3D_MODEL:
-                sp64 = 2;
+                objTypeMask = 2;
```

Comment:
question: is it reasonable to put these masks into an enum / defined constants? This looks like it should be something like
```c
#define UNK_MASK_2 = 1 << 1;
#define UNK_MASK_4 = 1 << 2;
```
And then reuse those masks (if they are re-used?)